mil_common/drivers/mil_passive_sonar/scripts/triggering

```diff
@@ -0,0 +1,207 @@
+
+        Subscribes to raw hydrophone samples on `/samples`
+
+        publishes samples from right around the triggering on `/pings` when a ping is detected
+            and found to be in the target frequency range
+
+        target frequency / tolerance(optional) must be set through the `/set_target` service call
+        '''
+
+        # Ros stuff
+        rospy.init_node('hydrophone_trigger')
+        self.sub = rospy.Subscriber('/samples', numpy_msg(HydrophoneSamples), self.hydrophones_cb)
+        self.pub = rospy.Publisher('/pings', numpy_msg(HydrophoneSamples), queue_size=1)
+        rospy.Service('/set_target', SetTargetFrequency, self.set_target)
+        rospy.Service('/enable', SetBool, self.enable)
+
+        self.enabled = True
+
+        # Attriburtes about our general frequency range (all pinger live here)
+        #  Frequency range garunteed to be relatively quiet except for the pingers (in Hz)
+        self.general_lower = 15000 # lowest frequency pinger - 10 kHz
+        self.general_upper = 50000 # highest frequency pinger + 10 kHz
+
+        # Attributes about our target frequency range
+        #  target Frquency in Hz
+        self.target = None
+        #  tolerance around that frequerncy in Hz
+        self.tolerance = 1000
+
+        # Filter attributes
+        #  Filter order for bandpass filter, (higher is better, but slower to compute)
+        self.filt_order = 201
+        #  Filter used to do a band pass filter for our general frquency range, calcuated on first cb
+        self.bandpass_filter = None
+        #  special filter property, do not change in Hz
+        self.trans_width = 1000
+
+        # Physical Properties
+        # min time between pings (of any kind) in sec
+        self.min_time_between_pings = 0.25
+        self.prev_trigger_time = -self.min_time_between_pings
+
+        # Debug attributes
+        #  Debug plot publishers
+        self.trigger_debug = Plotter('trigger_debug')
+        self.sample_at_trigger_debug = Plotter('sample_at_trigger_debug')
+
+        # Misc attributes
+        #  minimum gradient of the max convolution wrt time to trigger a time of arivals calcualtion
+        self.threshold = 40
+        #  expected error in determining exactly when the triggering happend (after) in sec
+        self.trigger_time_error_future = 0.0003
+        #  expected error in determining exactly when the triggering happend (before) in sec
+        self.trigger_time_error_past = 0.0003
+        # time (from start of samples)
+        self.time = 0.0
+        #  max convolution window, the amount of time that we assume should be quiet before a ping in sec
+        self.window_time = 100e-3
+        #  prev data buffer
+        self.prev_data = None
+
+
+    def enable(self, req):
+        self.enabled = req.data
+        res = SetBoolResponse()
+        res.success = True
+        return res
+
+
+    def set_target(self, req):
```

Comment:
So setting the target frequency doesn't affect the band pass? Wasn't the whole point of the redesign to have the bandpass change based on the freqency we're listening to?